### PR TITLE
docs: Voting ledger design

### PR DESCRIPTION
**Pull request by Mr-Leshiy**
# Description

Added initial design docs of the voting ledger

## Related Issue(s)

https://github.com/input-output-hk/catalyst-voices/issues/693

## Description of Changes

- Added design docs for voting ledger.
- Added document defined COSE algorithm tag values for BLAKE hash functions.

## Pages

 - https://input-output-hk.github.io/catalyst-voices/branch/docs_voting_ledger/architecture/08_concepts/immutable_ledger/ledger
 - https://input-output-hk.github.io/catalyst-voices/branch/docs_voting_ledger/catalyst-standards/cbor_tags/blake/

### CODE REVIEW

File: docs/src/architecture/08_concepts/immutable_ledger/cddl/block.cddl [hunk 1]
```diff
@@ -0,0 +1,32 @@
+block = [
+    block_header,
+    block_data: ~encoded-cbor, ; deterministically encoded CBOR
```
Comment (bkioshn) [inline docs/src/architecture/08_concepts/immutable_ledger/cddl/block.cddl:3]: From our discussion on Friday, should this be bytes string?

File: docs/src/architecture/08_concepts/immutable_ledger/cddl/block.cddl [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  #6.32781(bytes) \ ; Blake3 hash
+  #6.32782(bytes) \ ; Blake2b hash
+  #6.32783(bytes) ; Blake2s hash
+)
+kid = hash_bytes ; hash of the x509/c509 certificate
+
+validator = (kid / [2* kid])
+metadata = [ *Any ]
+
+block_data = ~encoded-cbor
```
Comment (bkioshn) [inline docs/src/architecture/08_concepts/immutable_ledger/cddl/block.cddl:31]: Redundant

File: docs/src/architecture/08_concepts/immutable_ledger/ledger.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+Project "Catalyst" requires a solution for storing people votes and any other data,
+in a transparent, verifiable, scalable and immutable way.
+
+## Specification
+
+### Ledger structure
+
+![Ledger schema](images/ledger_schema.svg){ align=right }
+
+Ledger will be represented as a collection of different, non-connected chains,
```
Comment (bkioshn) [inline docs/src/architecture/08_concepts/immutable_ledger/ledger.md:31]:
```suggestion
Ledger will be represented as a collection of distinct, unconnected chains,
```

File: docs/src/architecture/08_concepts/immutable_ledger/ledger.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+The only common thing for all these chains will be a "tree" identifier,
+so these chains will serve and form an overall ledger state.
+
+Obviously, given approach leads to data duplication,
+as each chain, will not know anything about others.
+And it also requires that the overall ledger state,
+could be deterministically defined at any point of time,
+considering potential transaction overlapping or duplication.
+
+Each particular chain, will be as a common sequence of blocks,
```
Comment (bkioshn) [inline docs/src/architecture/08_concepts/immutable_ledger/ledger.md:42]:
```suggestion
Each particular chain, will be a common sequence of blocks,
```

File: docs/src/architecture/08_concepts/immutable_ledger/ledger.md [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+<!-- markdownlint-disable no-inline-html -->
+<br clear="right"/>
+<!-- markdownlint-enable no-inline-html -->
+
+### Temporary chains
+
+![Temporary chain schema](images/temporary_chain.svg){ align=right }
+
+It is a common thing for blockchains to have a starting block (genesis),
+but it's unusual to have a final block for a chain.
```
Comment (bkioshn) [inline docs/src/architecture/08_concepts/immutable_ledger/ledger.md:56]: A bit picky,  but for consistency in the document, either one of  "it is" or "it's" should be used for the entire document.

File: docs/src/architecture/08_concepts/immutable_ledger/ledger.md [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+After which one no any block could be produced.
```
Comment (bkioshn) [inline docs/src/architecture/08_concepts/immutable_ledger/ledger.md:57]:
```suggestion
After which no any block could be produced.
```

**Review by bkioshn — COMMENTED** [review]
The document is cleared! just a minor issue 🥳

File: docs/src/architecture/08_concepts/immutable_ledger/cddl/block.cddl (hunk 1 above)
Comment (Mr-Leshiy) [reply docs/src/architecture/08_concepts/immutable_ledger/cddl/block.cddl:3]: But it is a bytes string, `encoded-cbor` is a type alias of `#6.24(bstr)`.
Here it is how it's defined in the cddl spec  `Appendix D.  Standard Prelude` https://datatracker.ietf.org/doc/rfc8610/
```cddl
 encoded-cbor = #6.24(bstr)
```

Comment (bkioshn) [reply docs/src/architecture/08_concepts/immutable_ledger/cddl/block.cddl:3]: Okey, understood 🙌🏼

**Review by cong-or — DISMISSED** [review]
LGTM

**Comment by stevenj** [comment]
Feedback from reviewer:
* The most major one: we probably want to elaborate on the "cryptographically protected by hashing" statement. What kind of attack are we trying to prevent? How is it sufficient?
* I failed to see how we identify the final block in the proposal. I may have read to quickly, but I'd say that either it's not specified or not emphasize enough.
* Many optional fields are optional and at the same time must have the same value as the previous block if present. It should probably not be the previous block but the previously defined value in the chain then.

**Review by stevenj — APPROVED** [review]
LGTM

**Pull request merged**